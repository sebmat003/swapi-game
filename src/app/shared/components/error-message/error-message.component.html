<div class="alert alert-danger mb-2" role="alert">
  {{ message }}
</div>
