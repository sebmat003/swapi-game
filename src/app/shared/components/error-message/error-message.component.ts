import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'app-error-message',
  templateUrl: './error-message.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ErrorMessageComponent {
  @Input() message = 'Error occured';
}
