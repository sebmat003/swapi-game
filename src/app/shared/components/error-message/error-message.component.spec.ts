import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { ErrorMessageComponent } from './error-message.component';

describe('ErrorMessageComponent', () => {
  let spectator: Spectator<ErrorMessageComponent>;
  const createComponent = createComponentFactory({
    component: ErrorMessageComponent,
    detectChanges: false,
  });

  beforeEach(() => (spectator = createComponent()));

  it('Should match snapshot', () => {
    spectator.detectChanges();

    expect(spectator.fixture).toMatchSnapshot();
  });
});
