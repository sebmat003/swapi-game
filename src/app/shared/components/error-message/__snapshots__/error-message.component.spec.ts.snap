// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessageComponent Should match snapshot 1`] = `
<app-error-message
  message={[Function String]}
>
  <div
    class="alert alert-danger mb-2"
    role="alert"
  >
     Error occured
  </div>
</app-error-message>
`;
